### ISSUE TITLE
dotnet runtime pack command finished leaving running dotnet.exe

### ISSUE THREAD

**Issue by AlexeiNaabal** [dotnet/runtime#74133]
### Description

OS: windows 10 19042.1706
Runtime: Product version: 6.0.9 , SdkBandVersion: 6.0.400

When invoking runtime/build.cmd and pass -pack command to generate a nuget package, the packaging finished successfully with several running dotnet.exe process, the number matches the logic cpu core number.
I tried start a new cmd process invoking command "dotnet build-server shutdown" right after the above return successfully, but the process hang after start, and those dotnet.exe still exists

Full arguments: (package procss) -pack -runtimeFlavor mono -a x64 -c Release /p:Deterministic=true
Full arguments: (cmd process after the process above return finished) dotnet build-server shutdown

Expecting packaging finished with these dotnet.exe terminated.

Is this a feature of runtime build procedural or a bug? Because it making our automation process unable to proceed by locking down the .dotnet folder and failing git clean, providing some way to turn this feature off is better

### Reproduction Steps

1. git clone runtime
2. switch to branch release/6.0
3. cd to /runtime
4. /build.cmd -pack -runtimeFlavor mono -a x64 -c Release /p:Deterministic=true , or any arch and configuration

### Expected behavior

No running dotnet.exe (describ as .NET Host) exists after packaging finished.

### Actual behavior

Several dotnet.exe running, the number matches the logic core number of cpu

**Comment by ViktorHofer** [comment]
> Several dotnet.exe running, the number matches the logic core number of cpu

AFAIK this by design. These could be the language server protocol nodes or msbuild nodes. @rainersigwald might know.

**Comment by akoeplinger** [comment]
Yeah but `dotnet build-server shutdown` should shut all of these server processes down and it looks like that hangs shutting down one or more of these processes.

I guess you could try shutting down individual servers, e.g. by running `dotnet build-server shutdown --msbuild` and `dotnet build-server shutdown --vbcscompiler`  and see if only one of them hangs.

**Comment by AlexeiNaabal** [comment]
> Yeah but `dotnet build-server shutdown` should shut all of these server processes down and it looks like it hangs shutting down one or more of these processes.
> 
> I guess you could try shutting down individual servers, e.g. by running `dotnet build-server shutdown --msbuild` and `dotnet build-server shutdown --vbcscompiler` and see if only one of them hangs.

Thanks for your info.

Our automation tool is a dotnet 6.0 console application itself. We invoke build.cmd command by creating new Process instance and wait for exit. I tried your suggestion in 2 ways:

1. Start 2 new Process right after previous pack Process return exit code, invoking: (1.1 => "dotnet build-server shutdown --msbuild") and (1.2 => "dotnet build-server shutdown --vbcscompiler"), after 1.1 start, the automation tool hangs at 1.1 => ProcessInstance.WaitForExit(), same if I switch 1.1 and 1.2. Also there are running dotnet.exe left. If I invoke this version of our tool from powershell, the console hangs too, right after 1.1 start.

2. From Powershell, I start our automation tool and pass pack command, no new process from 1. invoked, and the automation tool finished and quit. Then I use Invoke-Expression to invoke 1.1 and 1.2, there is server shutdown message printed on console, but those dotnet.exe does not terminated or exit. Script sequence like:

- Invoke-Expression "{our tool}.dll $Arguments" => success
- Invoke-Expression "dotnet build-server shutdown --msbuild" => success
- Invoke-Expression "dotnet build-server shutdown --vbcscompiler" => success
- Set-Location $DotNetRuntimePath
- git clean -fxd => failed here due to folder .dotnet is locked

**Comment by agocke** [comment]
I believe this is resolved. `build-server shutdown --all` appears to work for me.
